cpp/include/cudf/join.hpp

```diff
@@ -141,29 +191,59 @@ std::unique_ptr<cudf::table> inner_join(
  * @param[in] right_on The column indices from `right` to join on.
  * The column from `right` indicated by `right_on[i]` will be compared against the column
  * from `left` indicated by `left_on[i]`.
- * @param[in] columns_in_common is a vector of pairs of column indices into
- * `left` and `right`, respectively, that are "in common". For "common"
- * columns, only a single output column will be produced, which is gathered
- * from `left_on` columns. Else, for every column in `left_on` and `right_on`,
- * an output column will be produced.  For each of these pairs (L, R), L
- * should exist in `left_on` and R should exist in `right_on`.
  * @param[in] compare_nulls controls whether null join-key values
  * should match or not.
  * @param mr Device memory resource used to allocate the returned table and columns' device memory
  *
  * @return Result of joining `left` and `right` tables on the columns
- * specified by `left_on` and `right_on`. The resulting table will be joined columns of
- * `left(including common columns)+right(excluding common columns)`.
+ * specified by `left_on` and `right_on`.
  */
 std::unique_ptr<cudf::table> left_join(
   cudf::table_view const& left,
   cudf::table_view const& right,
   std::vector<cudf::size_type> const& left_on,
   std::vector<cudf::size_type> const& right_on,
-  std::vector<std::pair<cudf::size_type, cudf::size_type>> const& columns_in_common,
```

Comment:
This is listed as a non-breaking change, yet it's removing parameters on public APIs?